aioesphomeapi/api.proto

```diff
@@ -424,6 +424,8 @@ message ListEntitiesSensorResponse {
   int32 accuracy_decimals = 7;
   bool force_update = 8;
   string device_class = 9;
+  string state_class = 10;
+  string last_reset = 11;
```

Comment:
What is last_reset?

* If it's a dynamic attribute (changes across time), it should not be in this proto message. The List* messages are properties that describe the entity's static data, and for example in HA are persisted to disk.

* If this a datetime, it should be sent as an integer.

* It looks like this attribute is just there for some integrations like utility meter (correct me if i'm wrong). I don't think we should change ESPHome's sensor model just for making it work with one or two very Home Assistant specific integrations. For ESPHome, sensors only do one thing, and that is to represent a floating point value, and I think that simplicity is good. Keep in mind adding atteibutes to a core component like Sensor would also i crease the memory usage for every single integration (no matter if they use it or not).

Reply:
> If it's a dynamic attribute (changes across time), it should not be in this proto message. 

It can reset, for HA there is no difference. Please suggest an alterantive.

> If this a datetime, it should be sent as an integer.

It is an optional datetime, setting it to 0 is actually a valid one (and thus triggers logic), so hence I took this approach. An valid alternative is welcome.

> It looks like this attribute is just there for some integrations like utility meter (correct me if i'm wrong).

It is being implemented at this moment, it is for gathering long-term stats in the recorder (downsampling data).

> For ESPHome, sensors only do one thing, and that is to represent a floating point value, and I think that simplicity is good.

It signals Home Assistant to gather stats over sensors in the long term. Considering ESPHome provides a lot of sensors (just look at temperature), this really beneficial, otherwise, all ESPHome sensors would just be a current value displayed for the future.

Reply:
> It looks like this attribute is just there for some integrations like utility meter

That integration has been adjusted as well.

Reply:
Moved `last_reset` to `SensorState`

Reply:
> It is being implemented at this moment, it is for gathering long-term stats in the recorder (downsampling data).

I see; can you point me to where this is being implemented? I'm still not clear on both what this value is supposed to mean (what is "reset"? when the device/sensor last restarted?) and how that would impact downsampling (sensor values are just datapoints at a singular point in time and phyiscal sensors generally don't care when they were last reset, as they only give you the value of the current moment; I don't understand how/why a "reset" would matter for downsampling).

> It is an optional datetime, setting it to 0 is actually a valid one (and thus triggers logic), so hence I took this approach. An valid alternative is welcome.

I think 0 would be perfectly fine. In fact we could consider any datetime <2021-05-23 as illegal, because (as far as I understand the feature) such values can not happen without a bug. Alternatively, adding a second boolean that indicates the presence (like with light commands for example) would be ok, but only if there's a legitimate case where a dt of `0` would be valid.

> Moved last_reset to SensorState

Ok, that's better - depending on what the semantics of this property are, it could also be a device-wide property (all sensors are "reset" when the device restarts, and I can't think of a case where a sensor restarts without the device restarting).

Reply:
last_reset = 0 is epoch start, which is considered an non-resettable total. Thus not a correct value to send.

Reply:
> I see; can you point me to where this is being implemented? 

Quite a few PRs at the Home Assistant Core. I guess the best is to look at the PRs done by emontnemery last weeks on that repo.

Some key PRS:
- <https://github.com/home-assistant/core/pull/49852>
- <https://github.com/home-assistant/core/pull/50829>

Reply:
Converted value to an integer, setting the default to `-1`.